src/main/java/calculator/SimpleCalculator.java

```diff
@@ -0,0 +1,47 @@
+package calculator;
+
+import java.io.IOException;
+import java.security.InvalidParameterException;
+
+public class SimpleCalculator {
+
+    public static int sum(int a, int b) throws Exception {
+
+        long notIntChecker = ((long) a) + ((long) b);
```

Comment:
Too complicated casting mechanism. Wouldn't it possible to compare just integers? Or make the `sum` method arguments' type long if you wanted to do operations with long?

Reply:
In this case, I should cast int arguments to long, because it is impossible to check if there is an overflow by comparing only ints 
also, I want a method to accept ints and in order to check overflow I need to cast  them to long

Reply:
I got your point. However, even in that case type casting is a redundant operation. You might just subtract the first value from `Integer.MAX_VALUE` and check, whether it's bigger than the second value. 
Of course, this type of validation is good, but it's not complete. Yuor `sum` method can potentially get small negative values. In this case the sum of those two values will be out of Integer limit again, because it will be too small.
So, if you want to perform validation, you should do it from both sides.

Reply:
got it, thanks.